modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/TopLevelParameterObjectFactory.java

```diff
@@ -107,9 +110,11 @@ private void resolveParameters(ObjectType objectType, DefaultObjectBuilder build
       if (parameters.containsKey(key)) {
         valueResolver = toValueResolver(parameters.get(key));
       } else if (!isParameterGroup) {
-        valueResolver = getDefaultValue(field)
-            .map(value -> new TypeSafeExpressionValueResolver<>(value, objectField.getType(), muleContext))
-            .orElse(null);
+        if (getDefaultValue(field).isPresent()) {
```

Comment:
bceause of the prior comments, this code doesn't make sense anymore. and for the record, you could still do this with an Optional.map(), just put the if inside of it